jazon-core/src/main/java/com/zendesk/jazon/expectation/ArrayEachElementExpectation.java

```diff
@@ -0,0 +1,66 @@
+package com.zendesk.jazon.expectation;
+
+import com.zendesk.jazon.MatchResult;
+import com.zendesk.jazon.actual.*;
+import com.zendesk.jazon.mismatch.MismatchWithPath;
+import com.zendesk.jazon.mismatch.NullMismatch;
+import com.zendesk.jazon.mismatch.TypeMismatch;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+
+import java.util.List;
+
+import static com.zendesk.jazon.MatchResult.failure;
+import static com.zendesk.jazon.MatchResult.success;
+
+@RequiredArgsConstructor
+@EqualsAndHashCode
+class ArrayEachElementExpectation implements JsonExpectation {
+
+    private final JsonExpectation expectationForEachElement;
+
+    @Override
+    public MatchResult match(ActualJsonNumber actualNumber, String path) {
+        return failure(typeMismatch(ActualJsonNumber.class, path));
+    }
+
+    @Override
+    public MatchResult match(ActualJsonObject actualObject, String path) {
+        return failure(typeMismatch(ActualJsonObject.class, path));
+    }
+
+    @Override
+    public MatchResult match(ActualJsonString actualString, String path) {
+        return failure(typeMismatch(ActualJsonString.class, path));
+    }
+
+    @Override
+    public MatchResult match(ActualJsonNull actualNull, String path) {
+        return failure(
+                new NullMismatch<>(this)
+                        .at(path)
+        );
+    }
+
+    @Override
+    public MatchResult match(ActualJsonArray actualArray, String path) {
+        List<Actual> actualValues = actualArray.list();
+        for (int i = 0; i < actualValues.size(); ++i) {
```

Comment:
This may not be a big deal but generally it would be more safe to iterate using an `Iterator` interface. This is because `List.get()` does not guarantee constant time complexity. I would err on the side of caution here - just to be sure this will never be the reason to fix this part of code.